File: Example/cenarius/src/main/java/com/m/cenarius/Web/Interceptor/Interceptor.java
package com.m.cenarius.Web.Interceptor;

import android.content.Context;

import java.util.ArrayList;
import java.util.List;

/**
 * Created by m on 2017/5/10.
 */

public class Interceptor {

    public static final String scheme = "cenarius";

    private List<InterceptorAdapter> interceptors = new ArrayList<>();
    private static Interceptor sharedInstance = new Interceptor();

    public static void register(InterceptorAdapter interceptor) {
        sharedInstance.interceptors.add(interceptor);
    }

    public static boolean perform(String url, Context controller) {
        for (InterceptorAdapter interceptor: sharedInstance.interceptors) {
            if (interceptor.perform(url, controller)) {
                return true;
            }
        }
        return false;
    }
}

File: cenarius/src/main/java/com/m/cenarius/Web/Interceptor/Interceptor.kt
package com.m.cenarius.Web.Interceptor

import android.content.Context

import java.util.ArrayList

/**
 * Created by m on 2017/5/10.
 */

class Interceptor {

    private val interceptors = ArrayList<InterceptorAdapter>()

    companion object {

        val scheme = "cenarius"
        private val sharedInstance = Interceptor()

        fun register(interceptor: InterceptorAdapter) {
            sharedInstance.interceptors.add(interceptor)
        }

        fun perform(url: String, controller: Context): Boolean {
            for (interceptor in sharedInstance.interceptors) {
                if (interceptor.perform(url, controller)) {
                    return true
                }
            }
            return false
        }
    }
}
